rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,224 @@
+  );
+};
+
+MyComponent.propTypes = {
+  value: PropTypes.string,
+  onChange: PropTypes.function,
+  defaultValue: PropTypes.string,
+};
+
+export default MyComponent;
+```
+Here is a basic example of how a basic component will look in TS with uncontrolled support.
+
+```tsx
+// my-component.component.tsx
+interface MyComponentProps = {
+  value?: string;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  defaultValue?: string;
+};
+
+const MyComponent = ({ value, onChange, defaultValue, ...rest }: myComponentProps) => {
+  const isControlled = useRef(value !== undefined);
+  const inputRef = useRef(null);
+
+   const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (isControlled && onChange ) {
+      onChange(event);
+    } 
+  };
+
+  useEffect(() => {
+    // invariant goes in here.
+  }, [];
+
+  /* Component logic goes here */
+
+return (
+    <input 
+      value={value} 
+      onChange={handleOnChange} 
+      defaultValue={defaultValue} 
+      ref={inputRef}
+    />
+  )
+};
+
+export default MyComponent;
+```
+## Component Example - Controlled
+
+Here is a basic example of how a basic component will look in JS without uncontrolled.
+
+```jsx 
+// my-component.component.js
+const MyComponent = ({ value, onChange, ...rest }) => {
+   const handleOnChange = (event) => {
+    if (onChange) {
+      onChange(event);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+MyComponent.propTypes = {
+  value: PropTypes.string.isRequired,
+  onChange: PropTypes.function.isRequired,
+};
+
+export default MyComponent;
+```
+Here is a basic example of how a basic component will look in TS without uncontrolled.
+
+```tsx
+// my-component.component.tsx
+interface MyComponentProps = {
+  value: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+const MyComponent = ({ value, onChange, ...rest }: myComponentProps) => {
+   const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(event.target.value);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+export default MyComponent;
+```
+# Detailed design
+
+- We will add deprecation warnings to our input components so that our consumers are aware of the pending changes to remove uncontrolled support. This will appear as a console warning and a warning on our Storybook documentation.
+
+- We will remove uncontrolled behaviour from our input components and associated props piecemeal. Each of these will be a `BREAKING CHANGE`.
+
+This approach will provide immediate benefit to developers of Carbon as it means we will have less code and tests to maintain. It will also mean that consumers of Carbon can steadily upgrade their version of Carbon.
+
+# Components
+
+The input components that currently support being uncontrolled are: 
+1. Textbox
```

Comment:
`Switch`?

Reply:
@mkrds has added on the ones I've left out here https://github.com/Sage/carbon/pull/5029#discussion_r860627422